src/test/java/it/cnr/isti/hpc/wikipedia/article/en/ArticleTest.java

```diff
@@ -93,19 +95,116 @@ public void testNoEmptyAnchors() throws IOException {
 
         // No anchor should be empty
         for (Link link:a.getLinks()){
-            assertFalse(link.getDescription().isEmpty());
+            assertFalse(link.getAnchor().isEmpty());
         }
 
         // testing an specific anchor
         for (Link link:a.getLinks()){
             if (link.getId().equals("HTMS_Chakri_Naruebet"))
-                assertEquals(link.getDescription(),"HTMS Chakri Naruebet");
+                assertEquals(link.getAnchor(),"HTMS Chakri Naruebet");
         }
 
+    }
+    
+    @Test
+    public void testNoEmptyWikiIds() throws IOException {
+        Article a = new Article();
+        String mediawiki = IOUtils.getFileAsUTF8String("./src/test/resources/en/Cenozoic");
+        parser.parse(a, mediawiki);
 
+        for(Link l: a.getLinks()){
+            assert(!l.getId().equals(""));
+        }
+    }
+    
+    @Test
+    public void testEmptyLinksShouldBeFiltered() throws IOException {
```

Comment:
The code looks good, I like it. It will be compatible with the previous dumps and will add precious informations. Some notes: 
  - on my laptop a test fails, could you check if this happens also on yours? (I'm running them from the terminal with mvn test. I don't trust eclipse)
  - I run the compiled code on simple wikipedia English (simplewiki-20150702-pages-articles.json.gz) and in the records I found this one: 

   <code> 
  {
    "id": "April_27",
    "anchor": "April 27",
    "start": 5,
    "end": 5,
    "paragraphId": 0,
    "listId": 0,
    "listItem": 0,
    "tableId": 0,
    "rowId": 0,
    "colId": 0
  }
  </code> 

  I would use and Integer object set to null for all the *Id/ *Item so that if there's no value the    attribute will not appear in the json (0 as default in particular can be misleading), what do you think? I also noted that ``start`` is equal to ``end``    and that in this case type is not present (but other records have the type..)